pyfarm/agent/utility.py

```diff
@@ -392,3 +393,51 @@ def remove_file(
             raise
     else:
         logger.info("Removed %s", path)
+
+
+def remove_directory(
+        path, retry_on_exit=False, raise_=True, ignored_errnos=(ENOENT, )):
+    """
+    Simple function to **recursively** remove the provided directory or retry on
+    exit if requested.  This function standardizes the log output, ensures
+    it's only called once per path on exit and handles platform
+    specific exceptions (ie. ``WindowsError``).
+
+    :param bool retry_on_exit:
+        If True, retry removal of the file when Python exists.
+
+    :param bool raise_:
+        If True, raise an exceptions produced.  This will always be
+        False if :func:`remove_file` is being executed by :module:`atexit`
+
+    :param tuple ignored_errnos:
+        A tuple of ignored error numbers.  By default this function
+        only ignores ENOENT.
+    """
+    try:
+        shutil.rmtree(path)
+    except (WindowsError, OSError, IOError) as error:
+        if error.errno in ignored_errnos:
+            logger.debug(
```

Comment:
I think logging that is overkill. We are already producing enough log output that getting important information out of it is a lot of work.